src/core/ComboBox/ComboBox.tsx

```diff
@@ -380,84 +459,365 @@ export const ComboBox = <T,>(props: ComboBoxProps<T>) => {
       id={id}
     >
       <div className='iui-input-with-icon'>
-        <Popover
-          placement='bottom-start'
-          visible={isOpen}
-          onClickOutside={(_, { target }) => {
-            if (!toggleButtonRef.current?.contains(target as Element)) {
-              setIsOpen(false);
+        {children}
+        {endIcon}
+      </div>
+    </InputContainer>
+  );
+};
+ComboBoxInputContainer.displayName = 'ComboBoxInputContainer';
+
+const ComboBoxEndIcon = React.forwardRef(
+  (
+    props: React.ComponentPropsWithoutRef<'span'>,
+    forwardedRef: React.RefObject<HTMLSpanElement>,
+  ) => {
+    const { className, children, onClick: onClickProp, ...rest } = props;
+    const dispatch = useSafeContext(ComboBoxActionContext);
+    const { toggleButtonRef } = useSafeContext(ComboBoxRefsContext);
+    const refs = useMergedRefs(toggleButtonRef, forwardedRef);
+
+    return (
+      <span
+        ref={refs}
+        className={cx('iui-end-icon', className)}
+        onClick={(e) => {
+          onClickProp?.(e);
+          if (!e.isDefaultPrevented()) {
+            dispatch(['toggle']);
+          }
+        }}
+        {...rest}
+      >
+        {children ?? <SvgCaretDownSmall aria-hidden />}
+      </span>
+    );
+  },
+);
+ComboBoxEndIcon.displayName = 'ComboBoxEndIcon';
+
+const ComboBoxInput = React.forwardRef(
+  (props: InputProps, forwardedRef: React.RefObject<HTMLInputElement>) => {
+    const { onKeyDown: onKeyDownProp, onFocus: onFocusProp, ...rest } = props;
+
+    const { isOpen, id, focusedIndex } = useSafeContext(ComboBoxStateContext);
+    const dispatch = useSafeContext(ComboBoxActionContext);
+    const { inputRef, menuRef, optionsRef } = useSafeContext(
+      ComboBoxRefsContext,
+    );
+    const refs = useMergedRefs(inputRef, forwardedRef);
+
+    const focusedIndexRef = React.useRef({ original: -1, filtered: -1 });
+
+    // FIXME: Currently, the selected item is not focused on the first render.
+
+    // const forceRerender = React.useReducer(() => ({}), {})[1];
+    // const [optionsMounted, setOptionsMounted] = React.useState(false);
+    // React.useEffect(() => {
+
+    //   if (!isOpen) {
+    //     return;
+    //   }
+
+    //   if (isOpen && !menuRef.current) {
+    //     forceRerender();
+    //   }
+
+    //   if (menuRef.current?.children.length && optionsRef.current.length > 0) {
+    //     setOptionsMounted(true);
+    //   }
+    // });
+
+    React.useEffect(() => {
+      focusedIndexRef.current = {
+        original: focusedIndex ?? -1,
+        filtered: optionsRef.current.findIndex(
+          (el) => Number(el.dataset.iuiIndex) === focusedIndex,
+        ),
+      };
```

Comment:
`optionsRef` in deps will do nothing.

Reply:
Yeah, but the linter needs it because it's coming from context.